### DISCUSSION TITLE
RabbitMQ 3.8.0-1.e17 CRASH REPORT

### DISCUSSION THREAD

**Discussion by takazamy** [rabbitmq/rabbitmq-server#4205] [Questions]
My rabbit get CRASH REPORT

> 
> 2022-02-27 10:00:39 =CRASH REPORT====
>   crasher:
>     initial call: application_master:init/4
>     pid: <0.226.0>
>     registered_name: []
>     exception exit: {{bad_return,{{credentials_obfuscation_app,start,[normal,[]]},{'EXIT',{undef,[{crypto,hmac,[sha512,<<178,221,231,166,226,22,251,204,28,8,94,243,215,24,68,241,176,126,53,160,82,52,127,152,182,213,49,132,250,207,130,79,103,207,120,132,96,57,99,227,29,213,245,17,26,243,236,126,116,67,209,240,234,3,113,216,130,134,91,242,12,78,239,17,102,96,130,87,133,153,159,20,8,184,111,158,96,44,244,89,209,114,225,255,57,14,230,119,233,139,59,217,171,41,140,98,160,51,183,193,29,115,204,175,240,44,52,220,46,213,21,213,13,250,97,136,22,82,187,69,105,207,166,203,232,114,83,108>>,[<<45,45,17,83,153,111,14,129,142,192,206,166,247,74,135,84>>,<<0,0,0,1>>],64],[]},{credentials_obfuscation_pbe,xor_sum,7,[{file,"src/credentials_obfuscation_pbe.erl"},{line,272}]},{credentials_obfuscation_pbe,blocks,10,[{file,"src/credentials_obfuscation_pbe.erl"},{line,264}]},{credentials_obfuscation_pbe,make_key,5,[{file,"src/credentials_obfuscation_pbe.erl"},{line,121}]},{credentials_obfuscation_pbe,encrypt,5,[{file,"src/credentials_obfuscation_pbe.erl"},{line,106}]},{credentials_obfuscation_app,check,0,[{file,"src/credentials_obfuscation_app.erl"},{line,44}]},{credentials_obfuscation_app,start,2,[{file,"src/credentials_obfuscation_app.erl"},{line,36}]},{application_master,start_it_old,4,[{file,"application_master.erl"},{line,293}]}]}}}},[{application_master,init,4,[{file,"application_master.erl"},{line,142}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}
>     ancestors: [<0.225.0>]
>     message_queue_len: 1
>     messages: [{'EXIT',<0.227.0>,normal}]
>     links: [<0.225.0>,<0.44.0>]
>     dictionary: []
>     trap_exit: true
>     status: running
>     heap_size: 1598
>     stack_size: 28
>     reductions: 266
>   neighbours:
> 2022-02-27 10:00:39 =ERROR REPORT====
> Error in process <0.9.0> on node 'rabbit@prod-trumca-centos-1' with exit value:
> {undef,[{erlang,get_stacktrace,[],[]},{rabbit,boot_error,2,[{file,"src/rabbit.erl"},{line,1053}]},{rabbit,start_it,1,[{file,"src/rabbit.erl"},{line,468}]},{init,start_em,1,[]},{init,do_boot,3,[]}]}
> 2022-02-27 10:00:54 =CRASH REPORT====
>   crasher:
>     initial call: application_master:init/4
>     pid: <0.271.0>
>     registered_name: []
>     exception exit: {{bad_return,{{credentials_obfuscation_app,start,[normal,[]]},{'EXIT',{undef,[{crypto,hmac,[sha512,<<53,39,175,22,45,199,145,182,119,188,16,54,188,168,122,239,2,110,70,138,29,36,86,101,186,243,137,198,231,58,135,36,72,119,90,83,207,80,43,42,75,78,190,52,169,233,29,110,35,33,38,217,185,108,195,129,229,167,22,143,123,190,99,50,98,103,50,90,188,245,197,40,73,102,67,246,251,42,23,223,191,186,152,187,163,95,75,241,228,110,91,29,207,43,122,208,202,127,193,159,138,91,245,72,159,141,3,188,145,11,116,239,140,253,171,138,90,219,255,14,176,97,30,80,8,6,199,59>>,[<<154,68,87,128,7,88,140,135,153,126,171,15,30,80,75,30>>,<<0,0,0,1>>],64],[]},{credentials_obfuscation_pbe,xor_sum,7,[{file,"src/credentials_obfuscation_pbe.erl"},{line,272}]},{credentials_obfuscation_pbe,blocks,10,[{file,"src/credentials_obfuscation_pbe.erl"},{line,264}]},{credentials_obfuscation_pbe,make_key,5,[{file,"src/credentials_obfuscation_pbe.erl"},{line,121}]},{credentials_obfuscation_pbe,encrypt,5,[{file,"src/credentials_obfuscation_pbe.erl"},{line,106}]},{credentials_obfuscation_app,check,0,[{file,"src/credentials_obfuscation_app.erl"},{line,44}]},{credentials_obfuscation_app,start,2,[{file,"src/credentials_obfuscation_app.erl"},{line,36}]},{application_master,start_it_old,4,[{file,"application_master.erl"},{line,293}]}]}}}},[{application_master,init,4,[{file,"application_master.erl"},{line,142}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}
>     ancestors: [<0.270.0>]
>     message_queue_len: 1
>     messages: [{'EXIT',<0.272.0>,normal}]
>     links: [<0.270.0>,<0.44.0>]
>     dictionary: []
>     trap_exit: true
>     status: running
>     heap_size: 1598
>     stack_size: 28
>     reductions: 263
>   neighbours:
> 

I have no idea what is wrong, and how to fix it. Now my rabbitmq service cannot restart
In var/log/rabbitmq i get the log below

> 
> 2022-02-27 10:00:39.659 [error] <0.226.0> CRASH REPORT Process <0.226.0> with 0 neighbours exited with reason: call to undefined function crypto:hmac(sha512, <<178,221,231,166,226,22,251,204,28,8,94,243,215,24,68,241,176,126,53,160,82,52,127,152,182,213,...>>, [<<45,45,17,83,153,111,14,129,142,192,206,166,247,74,135,84>>,<<0,0,0,1>>], 64) in application_master:init/4 line 142
> 2022-02-27 10:00:39.659 [info] <0.44.0> Application credentials_obfuscation exited with reason: call to undefined function crypto:hmac(sha512, <<178,221,231,166,226,22,251,204,28,8,94,243,215,24,68,241,176,126,53,160,82,52,127,152,182,213,...>>, [<<45,45,17,83,153,111,14,129,142,192,206,166,247,74,135,84>>,<<0,0,0,1>>], 64)
> 2022-02-27 10:00:39.659 [info] <0.44.0> Application public_key exited with reason: stopped
> 2022-02-27 10:00:39.659 [info] <0.44.0> Application cowlib exited with reason: stopped
> 2022-02-27 10:00:39.660 [info] <0.44.0> Application crypto exited with reason: stopped
> 2022-02-27 10:00:39.662 [info] <0.44.0> Application sysmon_handler exited with reason: stopped
> 2022-02-27 10:00:39.667 [info] <0.44.0> Application mnesia exited with reason: stopped
> 2022-02-27 10:00:39.667 [info] <0.44.0> Application observer_cli exited with reason: stopped
> 2022-02-27 10:00:39.667 [info] <0.44.0> Application recon exited with reason: stopped
> 2022-02-27 10:00:39.667 [info] <0.44.0> Application xmerl exited with reason: stopped
> 2022-02-27 10:00:39.669 [info] <0.44.0> Application inets exited with reason: stopped
> 2022-02-27 10:00:39.669 [info] <0.44.0> Application asn1 exited with reason: stopped

**Comment by lukebakken** [comment]
Here is the error:

```
call to undefined function crypto:hmac
```

* What operating system are you using?
* How did you install RabbitMQ?
* What version of Erlang and RabbitMQ are you using?
* How did you install Erlang?

**Reply by takazamy** [reply]
- I'm using Centos 7
- I install RabbitMQ by sudo yum install rabbitmq-server-3.8.0-1.el7.noarch.rpm -y
- The verison Erlang i am not remember b/c the system setup a long time up till now
- The verison of RabbitMQ is 3.8.0-1e.17
- I install Erlang by: sudo yum install erlang -y
